### PR TITLE
Feature/spring io 17

### PR DESCRIPTION
**Pull request by Turbots**
Please checks for typos, content should be quite final, unless someone thinks otherwise ;-)

### CODE REVIEW

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+- UI Customization of all pages is possible through theming (*start drooling designers!*)
+
+All in all, the setup and demo went very smooth and **I genuinely feel this product is about to become very popular**, partly because of the Spring Boot integration, but also because it just seems very solid and user-friendly.
+There might be a dedicated blogpost coming soon about Keycloak, so stay tuned and check our blog regularly or subscribe to our [RSS feed](https://ordina-jworks.github.io/feed.xml){:target="_blank"}!
+
+# Spring Cloud Streams (Data Services)
+
+### by [Michael Minella](https://twitter.com/michaelminella){:target="_blank"}
+
+Michael gave a summary about all the new projects in the Spring ecosystem that process data and / or message very well.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: or messages

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+#### Solution: data microservices
+
+  - Developed and tested in isolation, also easier to test
+  - Independently scalable depending on data processing load
+  - Familiar development model, just like regular cloud-native Spring microservices
+  - Easier to govern for Ops
+  - So the need for data / app integration / composition arises
+  - Which means the need for orchestration and operational coverage arises (lots of plumbing required)
+
+#### Spring Cloud Stream
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: add link to the Spring Cloud project pages?
like: https://cloud.spring.io/spring-cloud-stream/

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+image: /img/springio2017.jpg
+tags: [Spring, IO, Pivotal, Google, Spring Boot, Java, Reactive, Reactor, DDD, CQRS, Spring Cloud, Spring Cloud Stream, Serverless]
+category: Spring
+comments: true
+---
+
+> On the 18th and 19th of May, we had another great edition of Spring I/O, brought to us by organizer [Sergi Almar](https://twitter.com/sergialmar){:target="_blank"}.
+In this blog post, we will go over some of our favourite sessions of the conference.
+
+> **DISCLAIMER**: this is obviously a very opinionated list of talks. This does not mean that the other talks were of lower quality, it just means we were not able to attend or did not take extensive notes.
```
Comment (MrJean) [inline _posts/2017-07-05-SpringIO17-Summary.md]: Sounds pretty negative. How about this?

"We were not able to attend all talks and therefore some pieces of this post are opinionated."

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 4]
```diff
@@ -0,0 +1,773 @@
...
+
+The keynote can be found [here](https://www.youtube.com/watch?v=uJSYnqIhcIo){:target="_blank"} and the demo code [here](https://github.com/snicoll-demos/demo-webflux-streaming){:target="_blank"}.
+
+# Bootiful Database-centric applications with JOOQ
+
+### by [Michael Simons](https://twitter.com/rotnroll666){:target="_blank"}
+
+<span class="image left small"><img class="p-image" alt="Michael Simons" src="/img/SpringIO2017/michael-simons.jpg"></span>
+
+As a personal fan of `@rotnroll666`, I went to see his talk, fully expecting to be impressed by the level of quality and code.
```
Comment (MrJean) [inline _posts/2017-07-05-SpringIO17-Summary.md]: Make Twitter handle clickable.

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 5]
```diff
@@ -0,0 +1,773 @@
...
+
+The biggest topic revolved around how **Spring has always tried to enable developers to adapt their applications rapidly**.
+This capacity of adapting to change increased dramatically when Spring Boot was released, which explains part of its success.
+
+They also reported the release of [Spring Boot 2.0.0 M1](https://spring.io/blog/2017/05/16/spring-boot-2-0-0-m1-available-now).
+This release was announced very shortly after [Spring Framework 5.0 went RC1](https://spring.io/blog/2017/05/08/spring-framework-5-0-goes-rc1) on the 8th of May.
+
+The keynote can be found [here](https://www.youtube.com/watch?v=uJSYnqIhcIo){:target="_blank"} and the demo code [here](https://github.com/snicoll-demos/demo-webflux-streaming){:target="_blank"}.
+
+# Bootiful Database-centric applications with JOOQ
```
Comment (MrJean) [inline _posts/2017-07-05-SpringIO17-Summary.md]: Although titles are capitalised, replace JOOQ with jOOQ. Also for all other occurrences.

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 6]
```diff
@@ -0,0 +1,773 @@
...
+
+# Reactive Spring Data
+
+### by Christophe Strobl
+
+#### Biggest changes of Spring Data Kay M3
+
+- Java 8 baseline
+- ID classes don't need to be Serializable anymore
+- **breaking change**: No more `null`s in the repositories (arguments or return  values)
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: the s of nulls does not fit with the layout

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 7]
```diff
@@ -0,0 +1,773 @@
...
+- Run the **code generator** to generate the Java DSL context (this happens in the Maven `generate-sources` lifecycle phase)
+- Use the DSL context to **write your typesafe queries**, for example:
+
+{% highlight java %}
+
+BookRecord book = create.selectFrom(BOOK).where(BOOK.ID.eq(1)).fetchOne();
+
+{% endhighlight %}
+
+For more information about jOOQ, you can check out their website [here](https://www.jooq.org/){:target="_blank"}.
```
Comment (MrJean) [inline _posts/2017-07-05-SpringIO17-Summary.md]: Rule number one for SEO purposes. Don't hide links behind "here" or "click here" alike things.

Better to create a single link, e.g.: "Read more about jOOQ and what you functionality it offers."

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 8]
```diff
@@ -0,0 +1,773 @@
...
+> Question: What's the difference between the jOOQ Query API and using JPA Criteria API with the Hibernate ModelGen, which is also typesafe?
+
+- It resembles the native SQL much better
+- jOOQ provides **standardization** since it performs SQL transformations that work for **any SQL dialect**
+- It should make it easier to migrate existing PL/SQL applications
+- There is a much more extensive collection of SQL functions and possibilities
+- jOOQ provides POJO mappers which are also generated from the code generator
+- As much as you hate them, it supports calling **Stored Procedures**!
+
+The code from Michael's talk can be found on [Github](https://github.com/michael-simons/bootiful-databases){:target="_blank"}.
```
Comment (MrJean) [inline _posts/2017-07-05-SpringIO17-Summary.md]: Same link thingy here: "Check out the code of Michael's talk on Github"

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 9]
```diff
@@ -0,0 +1,773 @@
...
+### Move towards SOA
+
+With Service-oriented Architecture the application is split up in components which can be deployed individually but now the key problem are the shared libraries.
+Releasing a change in an object that is not shared can be done separate, but still a problem for the shared ones.
+
+### Now we arrived at Microservices
+
+There are a lot of improvements but Microservices also adds the complexity of running a distributed system.
+Small, two pizza (5-8 members), teams organized around business capabilities with responsibility of running their own services.
+We gain independent deployability because each team produces and consumes rest api's to communicate.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: apis

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 10]
```diff
@@ -0,0 +1,773 @@
...
+    @Enumerated(EnumType.STRING)
+    private ProjectEventType type;
+    
+    private Project entity;
+    
+    private Map<String, Object> payload;
+
+{% endhighlight %}
+
+To make this way of working accessible to the developers, hypermedia api's need to expose links on the aggregates.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: apis

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 11]
```diff
@@ -0,0 +1,773 @@
...
+**Event handlers** are going to subscribe to an event and apply changes to an aggregate to change the state.
+In a graph representation, event handlers would be the nodes and events the edges.
+
+**CQRS** is used to create materialized views from streams of events.
+With CQRS you will have a command side and a query side.
+For example the commands might be written to an Apache Kafka event store.
+Then an event processor could be using Spring Cloud Stream to retrieve these events and create a data model.
+The data model is then written to a Data Store like MySQL, where the query side reads the data.
+An API gateway, like Spring Cloud Netflix Zuul, can be put in front so it looks for the consumer like a regular microservice.
+For deploying this application you can combine these components together of scale em independently.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: you can combine these components together of scale em independently =>  or scale them independently

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 12]
```diff
@@ -0,0 +1,773 @@
...
+We define functions that have a ServerRequest as parameter and we return a Mono with a ServerResponse.
+
+The request contains all the information we need.
+It contains the body of the request, pathvariables, request headers, ...
+So no more injecting pathvariables and body objects, we have everything we need in this ServerRequest.
+
+What we return is the ServerResponse in which we can easily put all the information we want to give back to the client.
+And Spring provides us with an easy builder to create such a response as it already did with the ResponseEntity builder.
+
+You can see that these new objects and builders provide us with an easy and declarative way to handle request and create responses, without the "magic" that we used previously with the annotations.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: to handle requests

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 13]
```diff
@@ -0,0 +1,773 @@
...
+    }
+    
+    public Mono<ServerResponse> listUsers(ServerRequest request) {
+        Flux<User> people = this.repository.allUsers();
+        return ServerResponse.ok().contentType(APPLICATION_JSON).body(users, User.class);
+    }
+}
+{% endhighlight %}
+
+Now we have defined how we want to handle request and how we translate it to a response.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: a request

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 14]
```diff
@@ -0,0 +1,773 @@
...
+What we need next is a way to say which requests will be handled by which handler function.
+
+In the old way this was done by defining an annotation that defined some parameters to couple for example a path to a method.
```
Comment (tomvdbulck) [inline _posts/2017-07-05-SpringIO17-Summary.md]: add commas to the 'for example' - or remove the 'for example'

File: _posts/2017-07-05-SpringIO17-Summary.md [hunk 15]
```diff
@@ -0,0 +1,773 @@
...
+And then in the main method we only have to start our Tomcat server and we're up and running:
+
+{% highlight java %}
+public static void main(String[] args) throws Exception {
+    Server server = new Server();
+    server.startTomcatServer();
+}
+{% endhighlight %}
+
+### Conclusion
```
Comment (MrJean) [inline _posts/2017-07-05-SpringIO17-Summary.md]: This is confusing, I thought I'd reached the end of the blog post.

File: _posts/2017-07-05-SpringIO17-Summary.md (hunk 3 above)
Comment (Turbots) [inline _posts/2017-07-05-SpringIO17-Summary.md]: hmm, dont think its that negative, its the truth :-)

**Comment by MrJean** [comment]
That doesn't mean you have to put it that way. It immediately raised doubt
about the quality of the post in my head. That's why I commented this.

It's all about perception!

2017-07-11 11:09 GMT+02:00 Dieter Hubau <notifications@github.com>:

File: _posts/2017-07-05-SpringIO17-Summary.md (hunk 7 above)
Comment (Turbots) [inline _posts/2017-07-05-SpringIO17-Summary.md]: cool, didnt know!

**Comment by MrJean** [comment]
Also, change aspect ratio of Gina.
And how about a real picture of Jeff?

**Comment by yannickdeturck** [comment]
As @MrJean said, I would leave out the DISCLAIMER since it doesn't really add anything.
We already have a disclaimer at the bottom of the site regarding everything being personal opinions and stuff: 
_Disclaimer: Opinions expressed on this blog reflect the writer's views and not the position of Ordina._

**Comment by Turbots** [comment]
I know, I created that disclaimer in the footer ;-)

But agreed, changing the disclaimer to something more useful

**Comment by Turbots** [comment]
Made all adjustments, should be okay now!

**Comment by MrJean** [comment]
@Turbots 

No index numbers in ToC?
No <hr> between articles?
Final conclusion?

**Pull request merged**